apps/portals/src/Navbar.tsx

```diff
@@ -223,7 +224,24 @@ class Navbar extends React.Component<any, State> {
                   >
                     <Modal.Header closeButton />
                     <SynapseComponents.Login
-                      sessionCallback={() => getSession()}
+                      twoFactorAuthenticationRequired={twoFaErrorResponse}
+                      onBeginOAuthSignIn={() => {
+                        // save current route (so that we can go back here after SSO)
+                        localStorage.setItem(
+                          'after-sso-login-url',
+                          window.location.href,
+                        )
```

Comment:
This behavior was always called when the Google OAuth button was clicked, but that's not appropriate for SWC, which doesn't use this key. Moved the logic here. Other client-only apps will need to follow-suite on SRC upgrade.

Reply:
At some point I do want to collect all of our client-only React app code into one utility, but haven't wanted to attempt that until we get another app into the monorepo